azure/services/scalesetvms/scalesetvms.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scalesetvms
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/converters"
+	infrav1exp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
+)
+
+type (
+	// ScaleSetVMScope defines the scope interface for a scale sets service.
+	ScaleSetVMScope interface {
+		logr.Logger
+		azure.ClusterDescriber
+		InstanceID() string
+		ScaleSetName() string
+		SetVMSSVM(vmssvm *infrav1exp.VMSSVM)
+		GetLongRunningOperationState() *infrav1.Future
+		SetLongRunningOperationState(future *infrav1.Future)
+	}
+
+	// Service provides operations on azure resources
+	Service struct {
+		Client client
+		Scope  ScaleSetVMScope
+	}
+)
+
+// NewService creates a new service.
+func NewService(scope ScaleSetVMScope) *Service {
+	return &Service{
+		Client: newClient(scope),
+		Scope:  scope,
+	}
+}
+
+// Reconcile idempotently gets, creates, and updates a scale set.
```

Comment:
This needs to be changed. Maybe something like "Reconcile fetches the latest data about the scaleset instance".